backend/api/server.js

```diff
@@ -408,13 +410,97 @@ server.get('/medications/due', async (req, res, next) => {
             }
         });
 
-        await reminderServiceUtils.updateMedicationDueReminders(medicationsDue)
         res.status(StatusCodes.OK).json(medicationsDue);
     } catch (e) {
         return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json(`Failed to retrieve medications due for reminders! Error: ${e.message}`);
     }
 });
 
+server.put('/medications/:medicationId/due', async (req, res, next) => {
+    const medicationId = Number(req.params.medicationId);
+    const medication = await prisma.medication.findUnique({
+        where: {
+            id: medicationId
+        }
+    });
+
+    if (!medication) {
+        return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json(`Failed to retrieve medication with id: ${medicationId}!`);
+    }
+
+    const patientId = medication.patient_id;
+    const patient = await prisma.patient.findUnique({
+        where: {
+            id: patientId
+        },
+        include: {
+            user: true,
+        }
+    })
+
+    const providerId = patient.provider_id;
+    const provider = await prisma.provider.findUnique({
+        where: {
+            id: providerId
+        },
+        include: {
+            user: true
+        }
+    })
+
+    await reminderServiceUtils.updateMedicationDueReminders(medication);
+
+    try {
+        const reminder = await prisma.sentReminders.create({
```

Comment:
Don't need to implement this, but a reminder that storing point-in-time info like this will scale your database usage very quickly. For other projects that need this, you can set up a cleanup job that deletes things older than 1 month or something